tck/src/main/java/org/eclipse/microprofile/concurrency/tck/cdi/JTACDITest.java

```diff
@@ -0,0 +1,310 @@
+                    Assert.assertEquals(currentValue + 1, transactionalService.getValue());
+                }).thenRunAsync(() -> {
+                    // the service call should have updated the bean in this transaction scope
+                    transactionalService.supports();
+                    Assert.assertEquals(currentValue + 2, transactionalService.getValue());
+                }).thenRunAsync(() -> {
+                    // updating a transaction scoped bean outside of a transacction should fail
+                    transactionalService.callServiceExpectFailure((tm, svc) -> svc.never(),
+                            transactionManager, transactionalService);
+                    Assert.assertEquals(currentValue + 2, transactionalService.getValue());
+                }).thenRunAsync(() -> {
+                    // updating a transaction scoped bean outside of a transacction should fail
+                    transactionalService.callServiceExpectFailure((tm, svc) -> svc.notSupported(),
+                            transactionManager, transactionalService);
+                    Assert.assertEquals(currentValue + 2, transactionalService.getValue());
+                }).thenRunAsync(() -> {
+                    transactionalService.mandatory();
+                    // the service call should have updated the bean in this transaction scope
+                    Assert.assertEquals(currentValue + 3, transactionalService.getValue());
+                });
+
+                stage0.join();
+                Assert.assertEquals(currentValue + 3, transactionalService.getValue());
+            }
+            finally {
+                transactionManager.rollback();
+            }
+        }
+        finally {
+            executor.shutdownNow();
+        }
+    }
+
+    @Test
+    // we don't yet have a precedent for how to signal lack of support for propagation to multiple threads in parallel,
+    @Ignore
+    /*
+     * Start two concurrent asynchronous stages and verify that
+     * a transaction scoped bean is updated twice.
+     */
+    public void testConcurrentTransactionPropagation() {
+        // create an executor that propagates the transaction context
+        ManagedExecutor executor = createExecutor("testConcurrentTransactionPropagation",
+                ThreadContext.TRANSACTION, ThreadContext.ALL_REMAINING);
+
+        if (executor == null) {
+            return; // the implementation does not support transaction propagation
+        }
+
+        try {
+            Assert.assertEquals(2, transactionalService.testConcurrentTransactionPropagation(
+                    executor, transactionManager), "testTransactionPropagation failed%n");
+            verifyNoTransaction();
+        }
+        finally {
+            executor.shutdownNow();
+        }
+    }
+
+    public void testRunWithTxnOfExecutingThread() throws SystemException, NotSupportedException {
```

Comment:
This test is missing the `@Test` annotation so it doesn't run.  If it did run, it would fail.  I think this test has a bug in that, if it aims to run under the transaction that is activate on the thread at the time when the task runs (as the method name suggests), the configuration should have `.propagated().unchanged(ThreadContext.TRANSACTION).cleared(ThreadContext.ALL_REMAINING)`

Reply:
Oops, and thanks for spotting the bug too.